asab/library/providers/filesystem.py

```diff
@@ -24,26 +24,21 @@ async def read(self, path):
 			return None
 
 
-	async def list(self, path):
-		recursive = path.endswith("*")
+	async def list(self, path, recursive=True):
 
 		if recursive:
-			path = os.path.join(path[:-2], "**")
+			path = os.path.join(path, "**")
 
-		file_names = glob.iglob(
-			os.path.join(self.LibraryBaseDir, path, "*.yaml"),
-			recursive=recursive
+		file_names = list(
+			glob.iglob(
+				os.path.join(self.LibraryBaseDir, path),
+				recursive=recursive
+			)
 		)
 
-		files_names_list = []
-
 		# Remove library path from the beginning of file names
 		library_path_to_replace = "{}/".format(os.path.abspath(self.LibraryBaseDir))
-
-		for file_name in file_names:
-			files_names_list.append(
-				file_name.replace(library_path_to_replace, '')
-			)
+		files_names_list = [i.replace(library_path_to_replace, '') for i in file_names]
```

Comment:
You want to strip the beginning of the path, not replace.

Reply:
I understand your point, but `strip` would bring more problems than `replace`. It does not strip the phrase but all characters found. I can append some weird characters at the beginning of the string and replace them afterwards, to be sure that I replace only the leading part. 
Some new string methods are in python 3.10 https://docs.python.org/3/library/stdtypes.html#str.removeprefix